File: app/src/main/java/com/prey/json/UtilJson.java
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2015 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.json;

import com.prey.PreyLogger;

import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class UtilJson {

    public static JSONObject makeJsonResponse(String command,String target,String status){
        JSONObject json=new JSONObject();
        try {
            json.put("command", command);
            json.put("target", target);
            json.put("status", status);
        } catch (JSONException e) {
            PreyLogger.e("Error:"+e.getMessage(),e);
        }
        return json;
    }

    public static JSONObject makeJsonResponse(String command,String target,String status,String reason){
        JSONObject json=makeJsonResponse(command, target, status);
        try {
            json.put("reason", reason);
        } catch (JSONException e) {
            PreyLogger.e("Error:"+e.getMessage(),e);
        }
        return json;
    }

    public static  Map<String, String> makeMapParam(String command,String target,String status){
        Map<String, String> map=new HashMap<String, String>();
        map.put("command", command);
        map.put("target", target);
        map.put("status", status);
        return map;
    }

    public static  Map<String, String> makeMapParam(String command,String target,String status,String reason){
        Map<String, String> map=makeMapParam(command, target, status);
        if(reason!=null)
            map.put("reason", reason);
        return map;
    }

    /**
     * Method get JSONObject
     *
     * @param jsonObject
     * @param key
     * @return value for key
     * @throws JSONException
     */
    public static JSONObject getJSONObject(JSONObject jsonObject, String key) throws JSONException {
        JSONObject out = null;
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getJSONObject(key);
        }
        return out;
    }

    /**
     * Method get string
     *
     * @param jsonObject
     * @param key
     * @return value for key
     * @throws JSONException
     */
    public static String getString(JSONObject jsonObject, String key) throws JSONException {
        String out = null;
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getString(key);
        }
        return out;
    }

    /**
     * Method get boolean
     *
     * @param jsonObject
     * @param key
     * @return value for key
     * @throws JSONException
     */
    public static boolean getBoolean(JSONObject jsonObject, String key) throws JSONException {
        boolean out = false;
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getBoolean(key);
        }
        return out;
    }

    /**
     * Method get int
     *
     * @param jsonObject
     * @param key
     * @return value for key
     * @throws JSONException
     */
    public static int getInt(JSONObject jsonObject, String key) throws JSONException {
        int out = 0;
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getInt(key);
        }
        return out;
    }

    /**
     * Method merge the json
     *
     * @param json1
     * @param json2
     * @return json merge
     */
    public static JSONObject merge(JSONObject json1, JSONObject json2) {
        try {
            Iterator<String> it = json2.keys();
            while (it.hasNext()) {
                String key = it.next();
                json1.put(key, json2.get(key));
            }
        } catch (Exception e) {
            PreyLogger.e(String.format("error merge:%s", e.getMessage()), e);
        }
        return json1;
    }

    /**
     * Method add json2 in json1 with key name
     *
     * @param json1
     * @param name
     * @param json2
     * @return json merge
     */
    public static JSONObject merge(JSONObject json1, String name, JSONArray json2) {
        try {
            json1.put(name, json2);
        } catch (Exception e) {
            PreyLogger.e(String.format("error merge:%s", e.getMessage()), e);
        }
        return json1;
    }

}

File: app/src/main/java/com/prey/json/UtilJson.kt
/*******************************************************************************
 * Created by Orlando Aliaga
 * Copyright 2025 Prey Inc. All rights reserved.
 * License: GPLv3
 * Full license at "/LICENSE"
 ******************************************************************************/
package com.prey.json

import com.prey.PreyLogger
import org.json.JSONException
import org.json.JSONObject

/**
 * Utility object for working with JSON data.
 */
object UtilJson {

    /**
     * Creates a JSON response with the given command, target, and status.
     *
     * @param command The command to include in the response.
     * @param target The target to include in the response.
     * @param status The status to include in the response.
     * @return A JSONObject representing the response.
     */
    fun makeJsonResponse(command: String?, target: String?, status: String?): JSONObject {
        val json = JSONObject()
        try {
            json.put("command", command)
            json.put("target", target)
            json.put("status", status)
        } catch (e: JSONException) {
            PreyLogger.e("Error:" + e.message, e)
        }
        return json
    }

    /**
     * Creates a JSON response with the given command, target, status, and reason.
     *
     * @param command The command to include in the response.
     * @param target The target to include in the response.
     * @param status The status to include in the response.
     * @param reason The reason to include in the response.
     * @return A JSONObject representing the response.
     */
    fun makeJsonResponse(
        command: String?,
        target: String?,
        status: String?,
        reason: String?
    ): JSONObject {
        val json = makeJsonResponse(command, target, status)
        try {
            json.put("reason", reason)
        } catch (e: JSONException) {
            PreyLogger.e("Error:" + e.message, e)
        }
        return json
    }

    /**
     * Creates a map of parameters with the given command, target, and status.
     *
     * @param command The command to include in the map.
     * @param target The target to include in the map.
     * @param status The status to include in the map.
     * @return A MutableMap of parameters.
     */
    fun makeMapParam(
        command: String,
        target: String,
        status: String?
    ): MutableMap<String, String?> {
        val paramMap = mutableMapOf(
            "command" to command,
            "target" to target,
            "status" to status
        )
        return paramMap
    }

    /**
     * Creates a map of parameters with the given command, target, status, and reason.
     *
     * @param command The command to include in the map.
     * @param target The target to include in the map.
     * @param status The status to include in the map.
     * @param reason The reason to include in the map.
     * @return A MutableMap of parameters.
     */
    fun makeMapParam(
        command: String,
        target: String,
        status: String?,
        reason: String?
    ): MutableMap<String, String?> {
        val paramMap = makeMapParam(command, target, status)
        reason?.let { paramMap["reason"] = it }
        return paramMap
    }

    /**
     * Retrieves a JSONObject from a given JSONObject based on a specified key.
     *
     * @param jsonObject The JSONObject to retrieve the value from.
     * @param key The key to look up in the JSONObject.
     * @return The JSONObject associated with the key, or null if not found.
     * @throws JSONException If the key is not found or the value is not a JSONObject.
     */
    @Throws(JSONException::class)
    fun getJSONObject(jsonObject: JSONObject?, key: String?): JSONObject? {
        var out: JSONObject? = null
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getJSONObject(key)
        }
        return out
    }

    /**
     * Retrieves a string value from a given JSONObject based on a specified key.
     *
     * @param jsonObject The JSONObject to retrieve the value from.
     * @param key The key to look up in the JSONObject.
     * @return The string value associated with the key, or null if not found.
     * @throws JSONException If the key is not found or the value is not a string.
     */
    @Throws(JSONException::class)
    fun getStringValue(jsonObject: JSONObject?, key: String?): String? {
        var out: String? = null
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getString(key)
        }
        return out
    }

    /**
     * Retrieves a boolean value from a given JSONObject based on a specified key.
     *
     * @param jsonObject The JSONObject to retrieve the value from.
     * @param key The key to look up in the JSONObject.
     * @return The boolean value associated with the key, or false if not found.
     * @throws JSONException If the key is not found or the value is not a boolean.
     */
    @Throws(JSONException::class)
    fun getBooleanValue(jsonObject: JSONObject?, key: String?): Boolean {
        var out = false
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getBoolean(key)
        }
        return out
    }

    /**
     * Retrieves an integer value from a given JSONObject based on a specified key.
     *
     * @param jsonObject The JSONObject to retrieve the value from.
     * @param key The key to look up in the JSONObject.
     * @return The integer value associated with the key, or 0 if not found.
     * @throws JSONException If the key is not found or the value is not an integer.
     */
    @Throws(JSONException::class)
    fun getIntValue(jsonObject: JSONObject?, key: String?): Int {
        var out = 0
        if (jsonObject != null && jsonObject.has(key)) {
            out = jsonObject.getInt(key)
        }
        return out
    }
}